4.x Use `Timer` base units to format `Timer#toString` and JSON output

tjquinno opened the pull request:
### Description
Resolves #9717 

## Release note
____
This change addresses two ways timer values are output:
* `MTimer#toString` continues to use `Duration` format but now shows as much precision as possible, regardless of any `baseUnit` setting on the timer.
* So users can control JSON output for timers, Helidon adds the new optional config setting `metrics.timers.json-units-default` which can have any `TimeUnit` value name.
   - If the config key is _absent_, Helidon expresses all timer values in seconds.
   - If the config key is _present_, Helidon uses the timer's explicit units setting if any was set and the config value otherwise.
   
   The default JSON output for timers in Helidon 4.x remains unchanged: JSON output for all timers is in seconds.
   
   For JSON output as in Helidon 3.x set `metrics.timers.json-units-default` to `NANOSECONDS`.
____

## Background
Helidon 4.x implements metrics using Micrometer, and although most Micrometer meters accept a `baseUnit` for storing the meter's values the `Timer` type does not. Instead, when code _retrieves_ a timer's value it specifies what Java `TimeUnit` the value should be scaled to. 

Helidon did not consistently deal with timer units.

## PR Overview

1. The `MTImer` class is the Helidon implementation of our `Timer` interface which delegates to Micrometer's `Timer`.
   * `MTimer` now stores the `baseUnit` setting if user code sets it and returns that value when asked. (It used to just use our superclass `MMeter#baseUnit` implementation which would return the `baseUnit` from the Micrometer delegate. Micrometer timers return seconds.)
   * `MTimer#toString` now retrieves the total time from the timer with as much precision as possible and (as before) formats it using `Duration`. (It used to retrieve the timer's `baseUnit` and retrieve the timer's value using that precision which was always seconds, leading to sometimes confusing output.)
   * Testing for this PR revealed an unrelated bug in the `Builder#buckets` method that is fixed.
2. The `MetricsConfigBlueprint` now has a new optional setting `metrics.timers.json-units-default`.
3. The `JsonFormatter` code now checks config for the new setting and computes the units to use for each timer accordingly (as described above).

### Documentation
Doc updates are included in the PR.

tjquinno commented:
Especially to reviewers @ljnelson @spericas @barchetta ...

This PR addresses an unintentional change in behavior from 3.x to 4.x. As posted now, it does so in a way that _preserves_ the 4.x behavior by default (for consistency among 4.x releases) while allowing users a way to switch back to the "correct" 3.x default behavior via config.

As I think more about this, I am inclined to change the solution so the default behavior would become the "correct" 3.x behavior, and if users want to preserve the "incorrect" 4.x behavior they would need to set a config value. But I'd like others' thoughts on that.

Again, this fix affects `MTimer#toString` output (which seems less problematic if it were to change between releases within 4) and also the JSON output from `/metrics` which seems as if it could be more of a potential problem--for users who might be depending on the existing "incorrect" JSON output.

spericas commented on metrics/api/src/main/java/io/helidon/metrics/api/MetricsConfigBlueprint.java:
```diff
@@ -250,6 +251,16 @@ static List<Tag> createTags(String pairs) {
+     */
+    @Option.Configured("timers.base-units-default")
+    Optional<TimeUnit> timerBaseUnitsDefault();
+
```
Why not provide a default (for this default) and dtop the optional here? Seems simpler and predictable.

spericas replied:
Not backward compatible maybe?

tjquinno replied:
Yes, that was pretty much my thinking. The current (before the PR) 4.x behavior is that the Helidon layer delegates to the Micrometer layer to decide what units to apply. Using an `Optional` here allows us to detect when the user config actually specified something so we know whether to use that or delegate to Micrometer.

spericas replied:
So Micrometer is not always using the same defaults?

tjquinno replied:
In typical (default) usage, Micrometer will use Prometheus metric implementations (including timer) but that's not guaranteed. Users could have Micrometer itself use some other implementation underneath, in which case the default behavior might be different. The Helidon code would not necessarily see that because it uses the Micrometer API, and our code does not look behind the scenes to see if the user has plugged in some other underlying implementation under Micrometer.

I'm not aware of any of our users actually doing this, but there's no reason they couldn't and no inherent reason we'd hear about it.

tjquinno commented:
Reviewers @spericas @ljnelson @barchetta 

After discussions with Joe I've revised the changes a bit. I've rewritten the PR description to reflect the slightly different approach and the changes made by the PR, and I've added a doc update.

barchetta reviewed (approved):
LGTM
